drafts/Stormless.md

```diff
@@ -0,0 +1,198 @@
+```
+
+
+Awesome! We can see that there is only one required parameter, `email`, but I'll add `first_name` to be conversational. The token can be passed as parameters, or as config. And if I choose to use config, my prior tribal knowledge hints that the `admin [object]` requires only `admin_token`. The very one I asked you to remember when you were setting up Slack workspace.
+> PRO TIP. While we are still polishing the plugin to expose all the Config details, you can find it out by exploring StackStorm Exchange pack config schema in `config.schema.yaml` file. For example, here is [config.example.yaml](https://github.com/StackStorm-Exchange/stackstorm-slack/blob/master/config.schema.yaml#L47-L78) for our Slack pack.
+
+Now we have all we need to create the heart of any Serverless project: the `serverless.yml`. Here it comes: 
+
+```
+service: signup-stormless
+
+provider:
+  name: aws
+  runtime: python2.7
+
+functions:
+  InviteSlack:
+    events:
+      - http:
+          method: POST
+          path: invite
+    stackstorm:
+      action: slack.users.admin.invite
+      input:
+        email:       "{{ input.body.email }}"
+        first_name:  "{{ input.body.first_name }}"
+      config:
+        admin: ${file(env.yml):slack}
+      output:
+        statusCode: 200
+        body: "{{ output }}"
+
+plugins:
+  - serverless-plugin-stackstorm
+```
+
+This is a good time to learn a bit of Serverless. The quick pointers below won't
+reading the docs to understand
+
+I threw in the `events` section here so that we can test the function with REST call to an AWS API Gateway endpoint.
+Note that this default configuration instructs API Gateway to pass the REST POST call with POST body under the `body` key ([details here](TODO-insert-link)). When we POST `{"first_name": "Santa", 
+"email": "santa@mad.russian.xmas.com"}`, the event passed to the Lambda is:
+
+```
+...
+"body": {
+    "first_name": "Santa", 
+    "email": "santa@mad.russian.xmas.com"
+}
+```
+Knowing the input data structure is important to map it to the action input parameters. It's intuitive: `input` represents `event` parameter of [AWS Lambda programming model](http://docs.aws.amazon.com/lambda/latest/dg/python-programming-model-handler-types.html) (BTW should we call it `event`? Vote with a PR!). [Jinja](http://jinja.pocoo.org/) is used to map the inputs; our JavaScript friends who're less familiar
+with this common Python tool find it intuitive enough in simple cases; and Stack-overflow is full of magic Jinga tricks.
+
+Here we obviously map the two parameters from input body to desired action input parameters. 
+
+To keep the config separately, I created a file `env.yml` and put my config parameters there:
+
+```
+# ./env.yml
+# WARNING: Don't commit to Github :)
+slack:
+  admin_token: "xoxs-111111111111-111111111111-111111111111-1111111111"
+  organization: "my-team"
+
+```
+Than I used it in `serverless.yml` like this: `admin: ${file(env.yml):slack}`. It's self-explanatory. 
```

Comment:
I prefer to stay away from language like "It's self-explanatory". Either explain it, or say nothing, IMO - what we take for granted may not always be obvious to someone else.